### PR TITLE
Profile page

### PR DESCRIPTION
**Pull request by fayeed**

### CODE REVIEW

File: src/components/Login.tsx [hunk 1]
```diff
@@ -35,9 +38,67 @@ const Login = (props: Props) => {
...
+    event.preventDefault();
+    const { value } = event.target;
+    setLastName(value);
+  }
+
   const generateAgent = async (event: React.SyntheticEvent) => {
     setLoading(true);
     let agentStatus = await ad4mClient.agent.generate(password);
+    const agentPerspective = await ad4mClient.perspective.add(
+      "My perspective"
```
Comment (kaichaosun) [inline src/components/Login.tsx]: Show we rename the name of perspective to `Agent Profile`?

File: src/components/Login.tsx [hunk 2]
```diff
@@ -35,9 +38,67 @@ const Login = (props: Props) => {
...
+    );
+    const links = [];
+
+    if (firstName) {
+      const link = await ad4mClient.perspective.addLink(
+        agentPerspective.uuid,
+        new Link({
+          source: 'ad4m://profile',
```
Comment (kaichaosun) [inline src/components/Login.tsx]: Can we make it a constant in a new file like `triples`, and other constant strings used?

File: src/components/Login.tsx [hunk 3]
```diff
@@ -35,9 +38,67 @@ const Login = (props: Props) => {
...
+      links.push(link)
+    }
+
+    const cleanedLinks = [];
+
+    for (const link of links) {
+      const newLink = JSON.parse(JSON.stringify(link));
+      newLink.__typename = undefined;
+      newLink.data.__typename = undefined;
+      newLink.proof.__typename = undefined;
```
Comment (kaichaosun) [inline src/components/Login.tsx:94]: I don't get why is these three lines needed, could you elaborate a bit?
I guess it's maybe required in GraphQL schema, can we move it to backend (ad4m executor) instead?

File: src/components/Login.tsx [hunk 4]
```diff
@@ -55,6 +116,24 @@ const Login = (props: Props) => {
     <div>
       <Stack align="center" spacing="xl">
         <div style={{ width: 280 }}>
+          {!isInitialized && (
+            <>
+              <TextInput 
+                label="First Name" 
+                placeholder='Jhon'
```
Comment (kaichaosun) [inline src/components/Login.tsx]: Do you mean `John`?

File: src/components/Login.tsx [hunk 5]
```diff
@@ -55,6 +116,24 @@ const Login = (props: Props) => {
...
+              <TextInput 
+                label="First Name" 
+                placeholder='Jhon' 
+                radius="md" 
+                size="md" 
+                onChange={onFirstNameChange}
+              />
+              <TextInput 
+                label="Last Name" 
+                placeholder='Doe'
```
Comment (kaichaosun) [inline src/components/Login.tsx]: Maybe placeholder for first name and last name to be `Satoshi` and `Nakamoto` 🦀  ?

File: src/components/Profile.tsx [hunk 6]
```diff
@@ -1,17 +1,133 @@
...
+            </ThemeIcon>
+          }
+        >
+          {trustedAgents.map((e, i) => (
+            <Card shadow="sm" withBorder={true} style={{ marginBottom: trustedAgents.length-1 === i ? 0 : 20 }}>
+              <Group align="flex-start">
+                <Avatar radius="xl"></Avatar>
+                <Group direction='column' style={{marginTop: 4}}>
+                  <Group  direction='row'>
+                    <Text weight="bold">Did: </Text>
```
Comment (kaichaosun) [inline src/components/Profile.tsx]:
```suggestion
                    <Text weight="bold">DID: </Text>
```

File: src/components/Profile.tsx [hunk 7]
```diff
@@ -1,17 +1,133 @@
...
+        <Button onClick={() => settrustedAgentModalOpen(true)}>Trusted Agents</Button>
+      </Container>
+      <Container
+        style={{ marginLeft: 30, marginTop: 62 }}
+      >
+        <Space h="md" />
+        <Text size="md" weight="bold" underline>Agent DID: </Text>
+        <Text size="md">{props.did}</Text>
+        <Space h="md" />
+        <Text size="md" weight="bold" underline>First Name: </Text>
```
Comment (kaichaosun) [inline src/components/Profile.tsx]: How about show full name directly?

File: src/components/Profile.tsx [hunk 8]
```diff
@@ -1,17 +1,133 @@
...
+
+  const [trustedAgentModalOpen, settrustedAgentModalOpen] = useState(false);
+  
+  const [profile, setProfile] = useState({
+    firstName: "",
+    lastName: ""
+  })
+
+  const getTrustedAgents = async () => {
+    const friends = await ad4mClient.runtime.friends()
```
Comment (kaichaosun) [inline src/components/Profile.tsx]: This is not used any more.

File: src/components/Profile.tsx [hunk 9]
```diff
@@ -1,17 +1,133 @@
...
+  }
+
+  const fetchProfile = async (agent: Agent) => {
+    const tempProfile = {
+      firstName: "",
+      lastName: ""
+    }
+
+    for (const { data: {source, predicate, target} } of agent.perspective?.links!) {
+      if (source === 'ad4m://profile') {
```
Comment (kaichaosun) [inline src/components/Profile.tsx]: This string can be reused when we have it as a constant.

File: src/components/Login.tsx (hunk 3 above)
Comment (fayeed) [reply src/components/Login.tsx:94]: yes, so we need to explicitly set it to undefined or the ad4mClient throw's an error as the type doesn't match

**Review by kaichaosun — APPROVED** [review]

**Pull request merged**